src/vcpkg/metrics.cpp

```diff
@@ -298,13 +214,11 @@ namespace vcpkg
             write_config = true;
         }
 
-#if defined(_WIN32)
-        if (config.user_mac.empty())
+        if (config.user_mac.empty() && LockGuardPtr<Metrics>(g_metrics)->metrics_enabled())
```

Comment:
This is in the `enable()` function itself -- it shouldn't check that the metrics are enabled.

Reply:
Fixed